x-pack/plugins/security/server/lib/flatten_object.ts

```diff
@@ -10,7 +10,7 @@ import { compact, isObject } from 'lodash';
 // Inspired by x-pack/plugins/observability_solution/apm/public/utils/flatten_object.ts
 // Slighly modified to have key/value exposed as Object.
 export const flattenObject = (
```

Comment:
Let's get rid of `any` completely i.e. use only `Record<PropertyKey, unknown>` everywhere as it's more correct (+ typo fix in the comment + improved `item[key]` access):
```ts
// Slightly modified to have key/value exposed as Object.
export const flattenObject = (
  item: Record<PropertyKey, unknown> | null | undefined,
  accDefault: Record<PropertyKey, unknown> = {},
  parentKey?: string
): Record<PropertyKey, unknown> => {
  if (item) {
    const isArrayWithSingleValue = Array.isArray(item) && item.length === 1;
    return Object.keys(item)
      .sort()
      .reduce<Record<PropertyKey, unknown>>((acc, key) => {
        const childKey = isArrayWithSingleValue ? '' : key;
        const currentKey = compact([parentKey, childKey]).join('.');
        // item[key] can be a primitive (string, number, boolean, null, undefined) or Object or Array
        // ⬆️ ISN'T IT "Captain Obvious"-like COMMENT? What else can item[key] be? :)
        const value = item[key];
        if (isObject(value)) {
          flattenObject(value as Record<PropertyKey, unknown>, acc, currentKey);
        } else {
          acc[currentKey] = value;
        }

        return acc;
      }, accDefault);
  }
  return {};
};
```

Reply:
😆 I like it. Fixed in latest commit